src/mon/ConfigMap.h

```diff
@@ -97,6 +97,14 @@ struct Section {
 };
 
 struct ConfigMap {
+  struct ValueSource {
+    std::string section;
+    const MaskedOption *option = nullptr;
```

Comment:
Can you add a comment re ownership of the MaskedOption here (or as a separate commit, as this one
is cherry-p'ed)?

Reply:
I have a follow-up PR that includes comments re each ConfigMap structs. I prefer to seprate the cherry-picked commits from the additional (non-trivial) ones I'll introduce.